components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarConsumer.java

```diff
@@ -76,4 +90,31 @@ protected void doResume() throws Exception {
         return strategy.create(endpoint);
     }
 
+    private Collection<ExecutorService> subscribeWithThreadPool(Collection<Consumer<byte[]>> consumers, PulsarEndpoint endpoint)
+            throws Exception {
+        int numThreads = endpoint.getPulsarConfiguration().getNumberOfConsumerThreads();
+        return consumers.stream().map(consumer -> {
+            ExecutorService executor = Executors.newFixedThreadPool(numThreads);
+            for (int i = 0; i < numThreads; i++) {
+                executor.submit(() -> {
+                    PulsarMessageListener listener = new PulsarMessageListener(endpoint, this);
+                    while (true) {
+                        try {
+                            Message<byte[]> msg = consumer.receive();
+                            listener.received(consumer, msg);
+                        } catch (PulsarClientException e) {
+                            if (e.getCause() instanceof InterruptedException) {
+                                // propagate interrupt
+                                LOGGER.info("Received shutdown signal, exiting");
+                                break;
+                            }
+                            LOGGER.error("Encountered exception while receiving message", e);
```

Comment:
Use getExceptionHandler to handle this exception.

Also you would need to catch Exception to have a general catch and handle to keep the thread up and running.